Pin SQLAlchemy to ``<1.3`` and update all dependencies

VJalili opened the pull request:
This PR does: 

- pins social-core and sqlalchemy versions;
- updates dependencies using `make update-dependencies`

ping @nsoranzo

VJalili committed:
Change cloudauthz version to 0.3.0.

VJalili committed:
Result of running `make update-dependencies`.

nsoranzo commented on lib/galaxy/dependencies/pipfiles/default/pinned-requirements.txt:
```diff
@@ -144,32 +146,32 @@ python-novaclient==11.0.0
 simplejson==3.16.0
 six==1.11.0
-social-auth-core[openidconnect]==3.1.0+gx0
+social-auth-core[openidconnect]==3.1.0
```
This was changed directly in this file in commit 5723e15cce8bc493a782b5aa18b894d230da6261 , so `make update-dependencies` is incorrectly reverting it. Can you apply:
```diff
diff --git a/lib/galaxy/dependencies/pipfiles/default/Pipfile b/lib/galaxy/dependencies/pipfiles/default/Pipfile
index 6741c92561..02777c6b6b 100644
--- a/lib/galaxy/dependencies/pipfiles/default/Pipfile
+++ b/lib/galaxy/dependencies/pipfiles/default/Pipfile
@@ -75,7 +75,7 @@ pyparsing = "*"
 "Fabric3" = "*"
 paramiko = "*"
 python-genomespaceclient = "<2.0"
-social_auth_core = {version = "==3.1.0", extras = ['openidconnect']}
+social_auth_core = {version = "==3.1.0+gx0", extras = ['openidconnect']}
 cloudauthz = "<=0.2.0"
 gxformat2 = "*"
 
```
to `lib/galaxy/dependencies/pipfiles/default/Pipfile` ? Or even pin it to the new 3.2.0 , if that works.

Then re-run `make update-dependencies`.

VJalili replied:
@nsoranzo reverted the change of social-auth-core; however, running `make update-dependencies` keeps dropping `+gx0` (and also updates `sqlalchemy` to its current latest version that causes some unrelated issues), so I did revert without running `make update-dependencies`. 

Similarly, running `make update-dependencies` reverts `social-auth-core[openidconnect]==3.2.0` to `3.1.0`. So, I left that version change to be addressed in a different PR.

nsoranzo replied:
Did you update the correct file `lib/galaxy/dependencies/pipfiles/default/Pipfile` before re-running `make update-dependencies` as in my comment above?
It did the right thing for me.
Also, it would be good to pin `sqlalchemy` to the current version in the same file (again, before re-running `make update-dependencies`).

nsoranzo replied:
I've pushed a commit implementing these to your branch.

VJalili replied:
Yes, my mistake, I've been changing a different file. Fixed it following your comment, but I guess we pushed at the same time :)  Thanks for fixing it.

VJalili committed:
Revert social-auth-core version change.

nsoranzo committed:
Pin SQLAlchemy and update social_auth_core pin

- SQLAlchemy 1.3 introduced changes that break some queries.

- Commit 5723e15cce8bc493a782b5aa18b894d230da6261 did not update the pin
  on `social_auth_core` in the Pipfile.

- Remove spurious `; python_version == '2.7'` produced by pipenv for
  `paste`.

- Run `make update-dependencies`.
